### DISCUSSION TITLE
Could you provide a minimal example demonstrating how to use otus to work with WASM?

### DISCUSSION THREAD

**Discussion by glyh** [yuriy-chumak/ol#50] [Q&A] [answered]
As title. 

I tried owl, first compile .scm to .c, then .c to .wasm with WASI-sdk, but it won't work.

So I tried this one, but seems like the CLI usage is different. 

Thanks a lot!

**Comment by yuriy-chumak** [comment, marked as answer]
Try `make olvm.wasm` from Otus Lisp repository folder. Emcc should be available by the $PATH (using `source path-to-the-emscripten/emsdk_env.sh`, for example).

You should be aware of a bug in emscripten for unblocking access to the stdin and other ports. Please check the Ol Makefile for details (line [304](https://github.com/yuriy-chumak/ol/blob/4c3bcca8db9a91337c2c01f8c9e046fe91414eb4/Makefile#L304), started from `# important note: fix for emscripten to asyncify stdin: ...`.

Additionally, you can check out the ["pacman" sample](https://github.com/yuriy-chumak/ol/tree/master/samples/pacman) to understood how to embed olvm into your code.

The full usage example can be found at the [project page sources](https://github.com/yuriy-chumak/ol/blob/gh-pages/javascripts/my.terminal.js) (sorry for lot of comments, I'm not a js dev so code is a bit shitty :)).
